Sources/Fluent/Query/QueryRepresentable.swift

```diff
@@ -107,6 +107,7 @@ extension QueryRepresentable where Self: ExecutorRepresentable {
 
         if let _ = entity.id, entity.exists {
             // update
+            try type(of: entity).willUpdate(entity: entity)
```

Comment:
You mean `Self.willUpdate(entity: Self)`? That doesn't work because that throws this error: `Type 'Self' has no member 'willUpdate'`, which makes sense, because I defined those methods on `Entity` and not `QueryRepresentable` nor `ExecutorRepresentable`.